docs/guides/getting-started/getting-started-scaling-and-tuning.adoc

```diff
@@ -0,0 +1,64 @@
+<#import "/templates/guide.adoc" as tmpl>
+<#import "/templates/links.adoc" as links>
+
+<@tmpl.guide
+title="Scaling"
+summary="Get started with {project_name} scaling and tuning">
+
+Once you have {project_name} started you can next consider adapting your instance to whatever load may be needed. Scaling and tuning can help you address:
+
+- minimizing resource utilization
+- excessive response times
+- database pool contention
+- out of memory errors, or excessive garbage collection overhead
+- providing higher availability via horizontal scaling
+
+== Vertical Scaling
+
+As you monitor your {project_name} workload if you see that CPU or memory is either under or over utilized, please consult <@links.ha id="concepts-memory-and-cpu-sizing" /> to better tune the resources available to the Java Virtual Machine (JVM).
+
+Before increasing the amount of memory available to the JVM, in particular when experiencing an out of memory error, it is best to determine what is contributing to the increased footprint via a heap dump. Excessive response times may also indicate the HTTP work queue is too large and tuning for load shedding would be better than simply providing more memory - see below.
+
+=== Tuning of the Java Virtual Machine
+
+Java Virtual Machine Tuning is highly related to the resources available. For example the default garbage collection settings may work well for nominal heap sizes, but may be not ideal for much smaller or larger JVMs. If you are experiencing excessive GC overhead in addition to utilizing the JVM garbage collection tuning options, please open an issue or discussion to raise awareness of your situation.
```

Comment:
JVM Option tuning is a bit vague, and IMHO it is missing how to set those options. AFAIK it is not possible to set those options in a supported way when using the Operator (only via a Pod Template). 

So it should either be more specific and supported, or removed.

Reply:
Yes it is unfortunately vague. Given that we've had quite a few issues of late related to this, it seems like we should mention it some way - perhaps just a reference to opening a discussion / issue based upon what symptoms are being experienced.

For the operator you can use an optimized image with environment variables set as needed - however we don't spell out this case anywhere else. Do you want to do that here, or would you rather pare this content down?

Reply:
Looking at the text again today, I think readers will have a hard time to identify a situation of "excessive GC overhead". 

In the KCB project we started to describe the available [JVM metrics](https://www.keycloak.org/keycloak-benchmark/kubernetes-guide/latest/running/jvm/jvm_metrics) and [JVM options](https://www.keycloak.org/keycloak-benchmark/kubernetes-guide/latest/running/jvm/jvm_options). still we haven't come far in describing

* problematic situations (like "excessive GC overhead"),
* how to identify them (by looking at the metrics), and 
* which remedies to take (by specifying JVM options). (cc: @kami619). 

So I would vote to remove or greatly reduce this paragraph, and create a follow-up issue that addresses all three items of the list above. I'd suggest to focus on one situation first (like "excessive GC overhead"), and possibly other situations later (like "out of memory). 

If you hold a different view on this, and consider having the text above is better than not having the text, I am ok to proceed and see what feedback we'll receive on it.

So the bottom line is, you're making the call here, and I'll be happy to approve it. 

BTW, this is the last paragraph we're discussing. All other comments are IMHO resolved.

Reply:
Ok, removed the JVM level text and opened #31376

Reply:
Thanks, this resolves this one.